source/dub/dependencyresolver.d

```diff
@@ -353,6 +294,12 @@ private string basePackage(string p)
 	return p[0 .. idx];
 }
 
+private string subPackage(string p)
+{
+	auto idx = indexOf(p, ':');
+	if (idx < 0) return "";
+	return p[idx+1 .. $];
```

Comment:
you could make this a one-liner using findSplit: `return p.findSplit(":")[2];`.

imo it's actually a lit bit more readable (and it would be even more readable if findSplit returned a tuple with start, match and end members) but it's not really required. Actually also think the performance might be a little bit worse.

Reply:
> it's actually a lit bit more readable (and it would be even more readable if findSplit returned a tuple with start, match and end members)

Yeah I tried this once

https://github.com/dlang/phobos/pull/5968

tl;dr: it was reverted due to a regression in CTFE usage of tuples.

Reply:
That would cause a range error for package names without a sub package part, wouldn't it?

Reply:
no findSplit returns a tuple, it is always exactly 3 long (pre, match, post) no matter the input

Reply:
Ah, I actually missed the "split" part.